sources/Launcher/DYNMarginCalculationLauncher.cpp

```diff
@@ -523,24 +531,50 @@ void MarginCalculationLauncher::findOrLaunchScenarios(const std::string& baseJob
   }
 
   for (unsigned int i=0; i < events2Run.size(); i++) {
-    double variation = events2Run[i].second;
+    double variation = events2Run[i].variation_;
     std::string iidmFile = generateIDMFileNameForVariation(variation);
     inputsByIIDM_.erase(iidmFile);  // remove iidm file used for scenario to save RAM
   }
 }
 
+MarginCalculationLauncher::EventPair::EventPair(size_t index, double variation) : index_(index), variation_(variation) {
+}
+
+void MarginCalculationLauncher::addAvailableEvents(const double variation,
+                                                   const std::vector<size_t>& eventsIdNoFilter,
+                                                   const std::vector<double >& maximumVariationPassing,
+                                                   std::vector<EventPair >& events2Run) {
+  if ((*loadIncreaseStatus_.find(variation)).second.success) {
+    auto found = scenarioStatus_.find(variation);
+    unsigned int eventIndex = 0;
+    while (events2Run.size() % mpi::context().nbProcs() != 0  && eventIndex < eventsIdNoFilter.size() &&
+           (variation > maximumVariationPassing[eventsIdNoFilter[eventIndex]] ||
+            DYN::doubleEquals(variation, maximumVariationPassing[eventsIdNoFilter[eventIndex]]))) {
+      EventPair event(eventsIdNoFilter[eventIndex], variation);
+      if (std::find(events2Run.begin(), events2Run.end(), event) == events2Run.end() && found == scenarioStatus_.end()) {
```

Comment:
Cannot we change the container type of events2Run to avoid this O(N) find?

Reply:
we could use an unordered_set maybe but as N is very small, as it is the number of procs, this find would not be very costly